enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1840,12 +1840,14 @@ type cdcWriter struct {
 	shouldCompress bool
 	isCompressed   bool
 
-	chunker       *chunker.Chunker
-	writtenChunks []*rspb.ResourceName
+	chunker         *chunker.Chunker
+	writtenChunks   []*rspb.ResourceName
+	writtenChunksMu *sync.Mutex // protects writtenChunks
 
-	numChunks  int
-	firstChunk []byte
-	fileType   rfpb.FileMetadata_FileType
+	numChunks   int
+	numChunksMu *sync.Mutex // protects numChunks
+	firstChunk  []byte
```

Comment:
it looks like firstChunk and fileType are also potentially written/read by multiple goroutines?